src/stores/search.ts

```diff
@@ -29,313 +30,282 @@ import {
   mediaFilterKeys,
   mediaUniqueFilterKeys,
 } from '~/constants/filters'
-import type { SupportedSearchType } from '~/constants/media'
-import type {
-  ApiQueryParams,
-  FilterCategory,
-  FilterItem,
-  Filters,
-} from '~/store/types'
 
-export const useSearchStore = defineStore('search', () => {
-  const state: {
-    searchType: SupportedSearchType
-    searchTerm: string
-    filters: Filters
-  } = reactive({
-    searchType: ALL_MEDIA,
-    searchTerm: '',
-    filters: clonedeep(filterData),
-  })
-  const { filters, searchType, searchTerm } = toRefs(state)
-
-  // Setters
-  const setSearchType = (type: SupportedSearchType) => {
-    state.searchType = type
-  }
-  function setSearchTerm(term: string) {
-    state.searchTerm = term.trim()
-  }
-  watch(searchType, (searchType) => clearOtherMediaTypeFilters(searchType))
+export interface SearchState {
+  searchType: SupportedSearchType
+  searchTerm: string
+  filters: Filters
+}
 
-  // Getters
-  /**
-   * Returns the search query parameters for API request:
-   * drops all parameters with blank values.
-   */
-  const searchQueryParams: ComputedRef<ApiQueryParams> = computed(() => {
-    return computeQueryParams(searchType.value, filters.value)
-  })
-
-  const computeQueryParams = (
-    searchType: SupportedSearchType,
-    filters: Filters = state.filters
-  ) => {
-    const query = { ...filtersToQueryData(filters, searchType) }
-
-    const queryKeys = Object.keys(query) as (keyof ApiQueryParams)[]
-    return queryKeys.reduce(
-      (obj, key: keyof ApiQueryParams) => {
-        if (key !== 'q' && query[key]?.length) {
-          obj[key] = query[key]
-        }
-        return obj
-      },
-      // Ensure that q filter always comes first
-      { q: state.searchTerm.trim() } as ApiQueryParams
-    )
-  }
+function computeQueryParams(
+  searchType: SupportedSearchType,
+  filters: Filters,
+  searchTerm: string
+) {
+  const query = { ...filtersToQueryData(filters, searchType) }
 
-  const allFilterCategories = Object.keys(state.filters) as FilterCategory[]
-  /**
-   * Initial filters do not include the provider filters. We create the provider filters object
-   * when we fetch the provider data on the Nuxt server initialization.
-   * We call this function to reset the filters to the initial base filters AND the provider filters.
-   */
-  const getBaseFiltersWithProviders = (): Filters => {
-    const resetProviders = (mediaType: SupportedMediaType): FilterItem[] => {
-      return state.filters[`${mediaType}Providers`].map((provider) => ({
-        ...provider,
-        checked: false,
-      }))
-    }
-    return {
-      ...clonedeep(filterData),
-      audioProviders: resetProviders(AUDIO),
-      imageProviders: resetProviders(IMAGE),
-    }
-  }
+  const queryKeys = Object.keys(query) as (keyof ApiQueryParams)[]
+  return queryKeys.reduce(
+    (obj, key: keyof ApiQueryParams) => {
```

Comment:
nit: This annotation isn't necessary, it's implicit in the type of `queryKeys`.

```suggestion
    (obj, key) => {
```